### PR TITLE
Don't stall coordinator init on slow remember shard store

### PR DESCRIPTION
**Pull request by johanandren**
Refs #28925

All sharding tests passing now but not quite confident I didn't miss some receive case in coordinator.

### CODE REVIEW

**Review by chbatey — APPROVED** [review]

**Comment by johanandren** [comment]
I'll go through the possible receives again tomorrow before I merge

**Comment by johanandren** [comment]
Gone through the possible receives now, will merge when green. 👍

**Pull request merged**
